client/java/src/main/java/com/linecorp/centraldogma/client/AbstractWatcher.java

```diff
@@ -329,11 +329,20 @@ private void doWatch(int numAttemptsSoFar) {
                                      projectName, repositoryName);
                          logged = true;
                      }
-                 }
+                 } else {
+                     if (cause instanceof CancellationException) {
+                         // Cancelled by close()
+                         return null;
+                     }
 
-                 if (cause instanceof CancellationException) {
-                     // Cancelled by close()
-                     return null;
+                     if (!initialValueFuture.isDone()) {
+                         // If the initial fetch failed with non-CentralDogmaException, it is likely that some
+                         // configuration is wrong or the file is malformed. So we complete the
+                         // initialValueFuture exceptionally rather than retrying indefinitely.
+                         initialValueFuture.completeExceptionally(thrown);
+                         close();
+                         return null;
+                     }
```

Comment:
Previously, it seems that if the network was temporarily unstable when the initial value had not been fetched, it would retry. However, with the new logic, it simply fails. Can this case be ignored, or should it be considered as well?

Reply:
On second thought, `RetryingClient` is not a good approach since Central Dogma watch uses long polling. `RetryingClient` will retain the references of requests for a long time. I switched to the implementation to retry up to 3 times when `initialValueFuture` is not completed.